tests/ui/codegen/empty-static-libs-issue-108825.rs

```diff
@@ -0,0 +1,15 @@
+//! Test that linking a no_std application still outputs the
+//! `native-static-libs: ` note, even though it's empty.
+
+//@ compile-flags: -Cpanic=abort --print=native-static-libs
+//@ build-pass
+//@ ignore-windows-msvc libcore links `/defaultlib:msvcrt` or `/defaultlib:libcmt`
+//@ ignore-cross-compile doesn't produce any output on i686-unknown-linux-gnu for some reason?
+
+#![crate_type = "staticlib"]
+#![no_std]
+
+#[panic_handler]
+fn panic(_info: &core::panic::PanicInfo) -> ! {
+    loop {}
+}
```

Comment:
Nit: can you add an

```rs
//@ error-pattern: note: native-static-libs:
```

to enforce that stderr must contain a "native-static-libs:" note, otherwise this can be easy to accidentally bless away.

Reply:
Actually wait let me check if we can use `//~` (super new) for this

Reply:
Also, does this test actually need the other note or specific contents of `note: native-static-libs:`? Because you can also dont-check-compiler-stderr, and instead rely on the error-pattern to assert the note still exists.

Reply:
Good suggestions! I've done that now, though have kept the `ignore-cross-compile` (I still don't know why that happens, I've tested `i686-unknown-linux-gnu` manually and it seems to produce the desired result there).